.test_director

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -xeuo pipefail
+
+NEW_TEST=$((git diff --name-only master | grep 'tests/.*/main.yml') || true)
```

Comment:
I think we can also end up in a state where a test is deleted.  Since we are not checking the status we could be trying to run a test on something that is no longer there.  Once I work out some of the current issues and have something working I'll add some more logic into it to handle different scenarios.